### PR TITLE
feat(UI): Add password policy form

### PR DESCRIPTION
**Pull request by Thebarda**
## Description

This adds a page that embed a password policy form.

https://user-images.githubusercontent.com/12515407/148612564-b3899cd7-eb07-4e60-8583-9a54744a62dc.mov

## Type of change

- [ ] Patch fixing an issue (non-breaking change)
- [x] New functionality (non-breaking change)
- [ ] Breaking change (patch or feature) that might cause side effects breaking part of the Software
- [ ] Updating documentation (missing information, typo...)

## Target serie

- [ ] 20.10.x
- [ ] 21.04.x
- [ ] 21.10.x
- [x] 22.04.x (master)

<h2> How this pull request can be tested ? </h2>

- Go to Administration > Authentication
- Updates some values in the form (don't forget to test edge cases provided by the corresponding example mapping)
- Click on the save button
- -> The form is correctly saved
- -> The "Save" and "Reset" buttons are now disabled
- Change a value
- Click on the "Reset" button
- Confirm the dialog
- -> Form is reset to initial values  

## Checklist

- [x] I have followed the **coding style guidelines** provided by Centreon
- [ ] I have commented my code, especially new **classes**, **functions** or any **legacy code** modified. (***docblock***)
- [ ] I have commented my code, especially **hard-to-understand areas** of the PR.
- [ ] I have made corresponding changes to the **documentation**.
- [x] I have **rebased** my development branch on the base branch (master, maintenance).

### CODE REVIEW

**Comment by bdauria** [comment]
It looks good, but I would add more spacing, both within the main container, and more generally, between the different elements

**Comment by Thebarda** [comment]
> It looks good, but I would add more spacing, both within the main container, and more generally, between the different elements
@bdauria Is it good ?
Screenshot 2021-12-03 at 14 23 31

**Comment by bdauria** [comment]
> > It looks good, but I would add more spacing, both within the main container, and more generally, between the different elements
> > @bdauria Is it good ?
> 
> Screenshot 2021-12-03 at 14 23 31

better to me 👍

**Review by a-launois — APPROVED** [review]

File: www/front_src/src/Authentication/Form/PasswordBlockingPolicy/BlockingDuration.tsx [hunk 1]
```diff
@@ -0,0 +1,120 @@
+import * as React from 'react';
+
+import { FormikValues, useFormikContext } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { isNil, not } from 'ramda';
+
+import { FormHelperText, FormLabel, useTheme } from '@material-ui/core';
+
+import { useMemoComponent } from '@centreon/centreon-frontend/packages/centreon-ui/src';
```
Comment (bdauria) [inline www/front_src/src/Authentication/Form/PasswordBlockingPolicy/BlockingDuration.tsx:9]:
```suggestion
import { useMemoComponent } from '@centreon/ui';
```

File: www/front_src/src/Authentication/LoadingSkeleton.tsx [hunk 2]
```diff
@@ -0,0 +1,78 @@
...
+  },
+  passwordExpirationAndBlockingGroups: {
+    display: 'flex',
+    flexDirection: 'column',
+    marginTop: theme.spacing(1),
+    rowGap: theme.spacing(1),
+  },
+}));
+
+const commonSkeletonProps: SkeletonProps = {
```
Comment (bdauria) [inline www/front_src/src/Authentication/LoadingSkeleton.tsx:35]: We could create a Component to avoid passing these props in every case

**Comment by jdelpierre** [comment]
So, Nice job for this form ! :) 

Why doesn't the white frame take all the space of the parent? visually I personally find it strange

**Comment by Thebarda** [comment]
> So, Nice job for this form ! :) 
> 
> Why doesn't the white frame take all the space of the parent? visually I personally find it strange 

It will be even more strange, like an emptiness sensation, with a huge white frame

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 3]
```diff
@@ -16314,3 +16314,123 @@ msgstr "Souple"
...
+msgid "Choose a duration between 1 hour and 12 months"
+msgstr "Veuillez choisir une durée entre 1 heure et 12 mois"
+
+msgid "Month"
+msgstr "Mois"
+
+msgid "Months"
+msgstr "Mois"
+
+msgid "Day"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16372]: already translated

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 4]
```diff
@@ -16314,3 +16314,123 @@ msgstr "Souple"
...
+msgstr "Jour"
+
+msgid "Days"
+msgstr "Jours"
+
+msgid "Hour"
+msgstr "Heure"
+
+msgid "Hours"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16381]: already translated

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 5]
```diff
@@ -16314,3 +16314,123 @@ msgstr "Souple"
...
+msgstr "Heures"
+
+msgid "Minute"
+msgstr "Minute"
+
+msgid "Minutes"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16387]: already translated

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 6]
```diff
@@ -16314,3 +16314,123 @@ msgstr "Souple"
...
+msgstr "Minutes"
+
+msgid "Second"
+msgstr "Seconde"
+
+msgid "Second"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16393]: `Seconds` is already translated

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po [hunk 7]
```diff
@@ -16314,3 +16314,123 @@ msgstr "Souple"
...
+msgid "Blocking duration must be less than or equal to 7 days"
+msgstr "La durée de blocage doit être inférieure ou égale à 7 jours"
+
+msgid "Saving..."
+msgstr "En cours de sauvegarde..."
+
+msgid "Saved"
+msgstr "Sauvegardé"
+
+msgid "Reset"
```
Comment (kduret) [inline lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16426]: already translated

File: lang/fr_FR.UTF-8/LC_MESSAGES/messages.po (hunk 6 above)
Comment (Thebarda) [reply lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16393]: But we need the singular here

Comment (kduret) [reply lang/fr_FR.UTF-8/LC_MESSAGES/messages.po:16381]: yes but you have a dupicate `msgid "Second"` :)

**Review by bdauria — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]

**Review by bdauria — APPROVED** [review]

**Review by jdelpierre — APPROVED** [review]
